### ISSUE TITLE
`Func<Task<int>>` disappears (version 1.3.6)

### ISSUE THREAD

**Issue by lsoft** [zompinc/sync-method-generator#52]
Hello.

The generator produces invalid sync version for the following async code

```C#
using OtherNamespace;

namespace SomeNamespace
{
    public partial class Subject1
    {
        [Zomp.SyncMethodGenerator.CreateSyncVersion]
        public async Task<int> FooAsync(
            Func<Task<int>> task
            )
        {
            var r = await task();
            return r;
        }
    }
}

```

![image](https://github.com/zompinc/sync-method-generator/assets/5988558/38d7e9ee-f702-4197-b836-fd319fbf335c)

**Comment by lsoft** [comment]
@virzak sorry, but are you sure it is fixed in 1.3.13? I can't confirm this...

**Comment by virzak** [comment]
@lsoft it will be in 1.3.16 momentarily.
